### PR TITLE
Add CollocationConstraint for handling volumes-from

### PR DESCRIPTION
**Pull request by sonchang**

### CODE REVIEW

**Comment by sonchang** [comment]
@alena1108 @ibuildthecloud feel free to review and merge (i guess after i squash)

**Comment by alena1108** [comment]
LGTM

File: code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/VolumesFromConstraintProvider.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    JsonMapper jsonMapper;
+
+    @Override
+    public void appendConstraints(AllocationAttempt attempt, AllocationLog log,
+            List<Constraint> constraints) {
+        Instance instance = attempt.getInstance();
+
+        if (instance != null) {
+            @SuppressWarnings("unchecked")
+            Set<Integer> dataVolumesFrom = DataAccessor.fields(instance).withKey(DockerInstanceConstants.FIELD_VOLUMES_FROM).as(jsonMapper, Set.class);
```
Comment (ibuildthecloud) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/VolumesFromConstraintProvider.java]: minor point, this should be a Long

Comment (sonchang) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/VolumesFromConstraintProvider.java]: I originally had it as a Long but ran into a ClassCastException and changed it

Comment (ibuildthecloud) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/VolumesFromConstraintProvider.java]: oh, then Number.  you can change it later.

**Pull request merged**